include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -693,34 +702,37 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
         }
         // TODO check if unordered_set might be faster
         // sort by id and increasing by distance
-        auto entry_point_comparator = [](const std::pair<NodeID, EdgeWeight> &lhs,
-                                         const std::pair<NodeID, EdgeWeight> &rhs) {
-            return lhs.first < rhs.first || (lhs.first == rhs.first && lhs.second < rhs.second);
-        };
-        std::sort(forward_entry_points.begin(), forward_entry_points.end(), entry_point_comparator);
-        std::sort(reverse_entry_points.begin(), reverse_entry_points.end(), entry_point_comparator);
+        std::sort(forward_entry_points.begin(), forward_entry_points.end());
+        std::sort(reverse_entry_points.begin(), reverse_entry_points.end());
+
+        // inserts the entry point in the heap if it wasn't inserted before and updates last_id
+        const auto insertInCoreHeap =
+            [](const CoreEntryPoint &p, SearchEngineData::QueryHeap &core_heap, NodeID &last_id) {
+                NodeID id;
+                EdgeWeight weight;
+                NodeID parent;
+                std::tie(id, weight, parent) = p;
+                if (id == last_id)
+                {
+                    return;
+                }
+
+                core_heap.Insert(id, weight, parent);
+                last_id = id;
+            };
 
-        NodeID last_id = SPECIAL_NODEID;
         forward_core_heap.Clear();
-        reverse_core_heap.Clear();
+        NodeID last_id = SPECIAL_NODEID;
         for (const auto &p : forward_entry_points)
         {
-            if (p.first == last_id)
-            {
-                continue;
-            }
-            forward_core_heap.Insert(p.first, p.second, p.first);
-            last_id = p.first;
+            insertInCoreHeap(p, forward_core_heap, last_id);
         }
+
+        reverse_core_heap.Clear();
         last_id = SPECIAL_NODEID;
         for (const auto &p : reverse_entry_points)
         {
-            if (p.first == last_id)
-            {
-                continue;
-            }
-            reverse_core_heap.Insert(p.first, p.second, p.first);
-            last_id = p.first;
+            insertInCoreHeap(p, reverse_core_heap, last_id);
```

Comment:
Good point. I can check the commit history on this, but I'm sure we added this later on because we were running into a bug regarding this. That might just have been a side-effect of some other problem long resolved.

If we don't even need to sort here that would be awesome - saves quite a bit overhead. I can remove it and we can see if the tests still pass.